### PR TITLE
Add binary redistributable project.

### PR DESCRIPTION
**Pull request by gtrevi**
## Description

Add binary redistributable project: the project will build a NuGet package containing the minimum required set of binaries for autonomously deploying the eBPF platform.
The packaged files are stored in the `\package\bin` folder.

## Testing

CI/CD

## Documentation

n.a.

### CODE REVIEW

File: tools/redist-package/ebpf-for-windows-redist.nuspec.in [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		<file src="ebpfapi.pdb" target="package\bin"/>
+		<file src="ebpfnetsh.dll" target="package\bin"/>
+		<file src="ebpfnetsh.pdb" target="package\bin"/>
+		<file src="ebpfforwindows.wprp" target="package\bin"/>
+		<file src="eBPFCore.sys" target="package\bin\drivers"/>
+		<file src="eBPFCore.pdb" target="package\bin\drivers"/>
+		<file src="NetEbpfExt.sys" target="package\bin\drivers"/>
+		<file src="NetEbpfExt.pdb" target="package\bin\drivers"/>
+		<file src="printk.sys" target="package\bin\drivers"/>
+		<file src="printk.pdb" target="package\bin\drivers"/>
```
Comment (dthaler) [inline tools/redist-package/ebpf-for-windows-redist.nuspec.in:35]: If this is "compact", why include printk.sys?

File: tools/redist-package/redist-package.vcxproj [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      <ConformanceMode>true</ConformanceMode>
+    </ClCompile>
+    <Link>
+      <SubSystem>Console</SubSystem>
+      <EnableCOMDATFolding>true</EnableCOMDATFolding>
+      <OptimizeReferences>true</OptimizeReferences>
+      <GenerateDebugInformation>true</GenerateDebugInformation>
+    </Link>
+  </ItemDefinitionGroup>
+  <!-- In accordance to what defined in sample.vcxproj, the build is disabled for the 'Analysis' CI/CD build, as it  does not generate the printk.sys artifact. -->
```
Comment (dthaler) [inline tools/redist-package/redist-package.vcxproj:82]:
```suggestion
  <!-- In accordance to what defined in sample.vcxproj, the build is disabled for the 'Analysis' CI/CD build, as it does not generate the printk.sys artifact. -->
```

Comment (dthaler) [reply tools/redist-package/redist-package.vcxproj:82]: Also, I can't parse "to what defined".  Maybe a missing "is"?

Comment (gtrevi) [reply tools/redist-package/redist-package.vcxproj:82]: yes, it's an implicit "is".

File: tools/redist-package/ebpf-for-windows-redist.nuspec.in (hunk 1 above)
Comment (gtrevi) [reply tools/redist-package/ebpf-for-windows-redist.nuspec.in:35]: As discussed internally, printk.sys was included as it'd useful as a quick test for loading a native program.
@shankarseal for comment on whether we want to stick to this decision or not.

File: .github/workflows/reusable-build.yml [hunk 3]
```diff
@@ -215,7 +215,19 @@ jobs:
       uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce
       with:
         name: ebpf-for-windows - NuGet package (${{inputs.build_artifact}}_${{env.BUILD_CONFIGURATION}})
-        path: ${{github.workspace}}/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/*.nupkg
+        path: ${{github.workspace}}/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/eBPF-for-Windows.*.nupkg
+
+    - name: Build the NuGet Redist package
```
Comment (shankarseal) [inline .github/workflows/reusable-build.yml:220]: Suggestion: Replace `Redist` by `Binary` ?

File: tools/redist-package/ebpf-for-windows-redist.nuspec.in (hunk 1 above)
Comment (shankarseal) [reply tools/redist-package/ebpf-for-windows-redist.nuspec.in:36]: Just a quick test if eBPF is working e2e. But I am OK with printk not being part of this package either.

**Review by shankarseal — APPROVED** [review]

File: tools/redist-package/ebpf-for-windows-redist.nuspec.in (hunk 1 above)
Comment (dthaler) [reply tools/redist-package/ebpf-for-windows-redist.nuspec.in:36]: I'd remove it then.

Comment (gtrevi) [reply tools/redist-package/ebpf-for-windows-redist.nuspec.in]: done.

File: .github/workflows/reusable-build.yml (hunk 3 above)
Comment (gtrevi) [reply .github/workflows/reusable-build.yml:220]: @shankarseal I kept the naming consistent with the project and artifacts (and resonance with other redistributables like the VC++ Redist), no problem in changing it across all if not ok.

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Comment by dthaler** [comment]
@gtrevi please rebase

**Comment by saxena-anurag** [comment]
@gtrevi it looks like one of the tests is failing (which may be unrelated to these changes). Maybe you can rebase your branch so that the latest changes to create crash dumps are pulled in -- this will ensure any future failure in CICD will create a dump to investigate

**Pull request merged**